src/install_rules.ml

```diff
@@ -184,25 +184,30 @@ module Gen(P : Install_params) = struct
       match lib.kind with
       | Normal | Ppx_deriver -> []
       | Ppx_rewriter ->
-        let pps = [(lib.buildable.loc, Pp.of_string lib.name)] in
-        let pps =
-          (* This is a temporary hack until we get a standard driver *)
-          let deps =
-            List.concat_map lib.buildable.libraries ~f:Lib_dep.to_lib_names
+        match (dir_kind : File_tree.Dune_file.Kind.t) with
+        | Dune ->
+          [Preprocessing.get_ppx_driver_for_public_lib sctx ~name ~dir_kind]
+        | Jbuild ->
+          let pps = [(lib.buildable.loc, Pp.of_string lib.name)] in
+          let pps =
+            let deps =
+              List.concat_map lib.buildable.libraries ~f:Lib_dep.to_lib_names
+            in
+            if List.exists deps ~f:(function
+              | "ppx_driver" | "ppx_type_conv" -> true
+              | _ -> false) then
+              pps @ [match Scope.name scope with
+                | Named "ppxlib" ->
+                  Loc.none, Pp.of_string "ppxlib.runner"
+                | _ ->
+                  Loc.none, Pp.of_string "ppx_driver.runner"]
+            else
+              pps
           in
-          if List.exists deps ~f:(function
-            | "ppx_driver" | "ppx_type_conv" -> true
-            | _ -> false) then
-            pps @ [match Scope.name scope with
-              | Named "ppxlib" ->
-                Loc.none, Pp.of_string "ppxlib.runner"
-              | _ ->
-                Loc.none, Pp.of_string "ppx_driver.runner"]
-          else
-            pps
-        in
-        let ppx_exe = Preprocessing.get_ppx_driver sctx ~scope pps in
-        [ppx_exe]
+          match Preprocessing.get_ppx_driver sctx ~scope ~dir_kind pps with
+          | Ok    x -> [x]
+          | Error _ ->
+            [Preprocessing.get_ppx_driver_for_public_lib sctx ~name ~dir_kind]
```

Comment:
Could you explain this bit? What kind of errors are we ignoring here?

Reply:
Looking at the code again, I believe the only possible failure is if `ppxlib.runner` or `ppx_driver.runner` doesn't exist. This might be the case at some point in the future and this code is not very clear anyway. I reworked it to make things more explicit and avoid the error case here.